gc.c

```diff
@@ -11667,7 +11667,7 @@ rb_raw_obj_info(char *buff, const int buff_size, VALUE obj)
             }
 	    break;
 	  case T_STRING: {
-            APPENDF((BUFF_ARGS, "%s", RSTRING_PTR(obj)));
+            APPENDF((BUFF_ARGS, "%s", StringValueCStr(obj)));
```

Comment:
Instead of trying to convert the Ruby string to a C string, why not use the length directly? 
```suggestion
            APPENDF((BUFF_ARGS, "%.*s", RSTRING_LEN(obj), RSTRING_PTR(obj)));
```
StringValueCStr could raise which is could be problematic for this function.